include/_antic.h

```diff
@@ -33,66 +44,206 @@
 #ifndef __ANTIC_H
 #define __ANTIC_H
 
-/* Define a structure with the antic register offsets */
+/*****************************************************************************/
+/* Define a structure with the ANTIC coprocessor's register offsets          */
+/*****************************************************************************/
+
 struct __antic {
-    unsigned char   dmactl; /* direct memory access control */
-    unsigned char   chactl; /* character mode control */
+    unsigned char   dmactl; /* (W) direct memory access control */
+    unsigned char   chactl; /* (W) character mode control */
     unsigned char   dlistl; /* display list pointer low-byte */
     unsigned char   dlisth; /* display list pointer high-byte */
-    unsigned char   hscrol; /* horizontal scroll enable */
-    unsigned char   vscrol; /* vertical scroll enable */
+    unsigned char   hscrol; /* (W) horizontal scroll enable */
+    unsigned char   vscrol; /* (W) vertical scroll enable */
     unsigned char   unuse0; /* unused */
-    unsigned char   pmbase; /* msb of p/m base address */
+    unsigned char   pmbase; /* (W) msb of p/m base address (for when DMACTL has player and/or missile DMA enabled) */
     unsigned char   unuse1; /* unused */
-    unsigned char   chbase; /* character base address */
-    unsigned char   wsync;  /* wait for horizontal synchronization */
-    unsigned char   vcount; /* vertical line counter */
-    unsigned char   penh;   /* light pen horizontal position */
-    unsigned char   penv;   /* light pen vertical position */
-    unsigned char   nmien;  /* non-maskable interrupt enable */
-    unsigned char   nmires; /* nmi reset/status */
+    unsigned char   chbase; /* (W) msb of character set base address */
+    unsigned char   wsync;  /* (W) wait for horizontal synchronization */
+    unsigned char   vcount; /* (R) vertical line counter */
+    unsigned char   penh;   /* (R) light pen horizontal position */
+    unsigned char   penv;   /* (R) light pen vertical position */
+    unsigned char   nmien;  /* (W) non-maskable interrupt enable */
+    unsigned char   nmires;
+    /* (W) ("NMIRES") nmi reset -- clears the interrupt request register; resets all of the NMI status together
+    ** (R) ("NMIST") nmi status -- holds cause for the NMI interrupt
```

Comment:
You could use an anonymous union here to provide both `nmist` and `nmires`, like
```
union {
  unsigned char nmires;  /* (W) ... .*/
  unsigned char nmist;   /* (R) .... */
}
```

Reply:
Nice! I admit I've not had any reason to use `union` since, like, college 25 years ago! :smile: